### PR TITLE
Add Gunicorn with no version fingerprint

### PR DESCRIPTION
**Pull request by mkienow-r7**
## Description
Adds fingerprint for Gunicorn with no version.

## Motivation and Context
Improved coverage

## How Has This Been Tested?
* `bundle exec ./bin/recog_verify xml/http_servers.xml`
* `rake tests`

## Types of changes

- New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have updated the documentation accordingly (or changes are not required).
- [x] I have added tests to cover my changes (or new tests are not required).
- [x] All new and existing tests passed.

### CODE REVIEW

File: xml/http_servers.xml [hunk 1]
```diff
@@ -957,14 +957,22 @@
   </fingerprint>
 
   <fingerprint pattern="^gunicorn\/([\d.]+)+$">
-    <description>Gunicorn Gunicorn</description>
+    <description>Gunicorn with version</description>
     <example service.version="19.7.1">gunicorn/19.7.1</example>
```
Comment (cbarnard-r7) [inline xml/http_servers.xml:961]: Similar version issue with multiple/trailing periods.

`^gunicorn\/(\d+(.\d+)*)+$`

Again, not super crucial.

**Review by cbarnard-r7 — COMMENTED** [review]
Curious about your thoughts on version strings.

File: xml/http_servers.xml (hunk 1 above)
Comment (mkienow-r7) [reply xml/http_servers.xml:961]: I've modified the version pattern to `(\d+(?:\.\d+)*)` that allows a single version with no period character and also prevents matching version strings that contain multiple period characters in sequence.

```
$ echo "gunicorn/19" | bundle exec ./bin/recog_match xml/http_servers.xml
MATCH: {"matched"=>"Gunicorn with version", "service.vendor"=>"Gunicorn", "service.product"=>"Gunicorn", "service.version"=>"19", "service.cpe23"=>"cpe:/a:gunicorn:gunicorn:19", "service.protocol"=>"http", "fingerprint_db"=>"http_header.server", "data"=>"gunicorn/19"}

$ echo "gunicorn/19.7" | bundle exec ./bin/recog_match xml/http_servers.xml
MATCH: {"matched"=>"Gunicorn with version", "service.vendor"=>"Gunicorn", "service.product"=>"Gunicorn", "service.version"=>"19.7", "service.cpe23"=>"cpe:/a:gunicorn:gunicorn:19.7", "service.protocol"=>"http", "fingerprint_db"=>"http_header.server", "data"=>"gunicorn/19.7"}

$ echo "gunicorn/19.7.1" | bundle exec ./bin/recog_match xml/http_servers.xml
MATCH: {"matched"=>"Gunicorn with version", "service.vendor"=>"Gunicorn", "service.product"=>"Gunicorn", "service.version"=>"19.7.1", "service.cpe23"=>"cpe:/a:gunicorn:gunicorn:19.7.1", "service.protocol"=>"http", "fingerprint_db"=>"http_header.server", "data"=>"gunicorn/19.7.1"}

$ echo "gunicorn/19.7...1" | bundle exec ./bin/recog_match xml/http_servers.xml
FAIL: gunicorn/19.7...1
```

**Review by cbarnard-r7 — APPROVED** [review]
LGTM

**Pull request merged**